examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,157 @@
+                          LinearImplicitSystem &system,
+                          std::string var_name, bool needs_re_init =false) {
+    system.add_variable(var_name, CONSTANT, MONOMIAL);
+    if (!needs_re_init){
+        equation_systems.init();
+    }
+    else{
+        equation_systems.reinit();
+    }
+}
+
+double GetElementDataFromMesh(LinearImplicitSystem &system,const libMesh::Elem& elem,
+                                    const unsigned int variable_num){
+
+    DofMap &dof_map = system.get_dof_map();
+    std::vector<dof_id_type> dof_indices;
+    dof_map.dof_indices(&elem, dof_indices,variable_num);
+
+    std::vector<double> solution_value(1);
+    system.solution->get(dof_indices, solution_value);
+
+    return static_cast<double> (solution_value[0]);
+}
+
+const unsigned int FindCluster(libMesh::Mesh &mesh, LinearImplicitSystem &system,
+                 const std::string & variable_name) {
+
+  const unsigned int variable_num = system.variable_number(variable_name);
+
+  const unsigned int index = mesh.add_elem_integer(variable_name);
+
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    int element_solution = static_cast<int>(GetElementDataFromMesh(system,*elem,variable_num));
+
+    bool belong_to_a_cluster = false;
+    for (unsigned int side = 0; side < elem->n_sides(); ++side) {
+      const Elem *neighbor = elem->neighbor_ptr(side);
+      if (neighbor) {
+
+        int neighbor_element_solution = static_cast<int>(GetElementDataFromMesh(system,*neighbor,variable_num));
+
+        if (element_solution == neighbor_element_solution) {
+          belong_to_a_cluster = true;
+          //Danger Zone for future ref
+          //cluster_id should never be equal to element_solution
+          unsigned int cluster_id =element_solution;
+          elem->set_extra_integer(index,cluster_id);
+          break;
+        }
+      }
+    }
+    if (!belong_to_a_cluster){
+        elem->set_extra_integer(index,0);
+    }
+  }
+  return index;
+
+}
+
+void CaptureClusterID(libMesh::Mesh & mesh, libMesh::EquationSystems &equation_systems,const unsigned int index){
+
+    LinearImplicitSystem &local_system = equation_systems.add_system<LinearImplicitSystem>("cluster_id");
+    DofMap &local_dof_map = local_system.get_dof_map();
+    AddVariablesToSystem(equation_systems, local_system, "cluster_id_field",true);
+
+    for (const auto & elem:mesh.element_ptr_range()){
+        std::vector<dof_id_type> local_dof_indices;
+        local_dof_map.dof_indices(elem, local_dof_indices);
+        int cluster_id = elem->get_extra_integer(index);
+        local_system.solution->set(local_dof_indices[0], cluster_id);
+    }
+
+    local_system.solution->close();
+
+}
+
+void CloseSystems(libMesh::Mesh &mesh, LinearImplicitSystem &system,
+                  libMesh::EquationSystems &equation_systems,
+                  bool print_system_info = true) {
+  system.solution->close();
```

Comment:
I think this is already closed in `PopulateRandomIntegers()`?

Do we need this function?